pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -39,42 +48,89 @@ func (*AgentClusterInstall) Dependencies() []asset.Asset {
 
 // Generate generates the AgentClusterInstall manifest.
 func (a *AgentClusterInstall) Generate(dependencies asset.Parents) error {
-	// installConfig := &agent.OptionalInstallConfig{}
-	// dependencies.Get(installConfig)
-
-	// agentClusterInstall := &hiveext.AgentClusterInstall{
-	// 	ObjectMeta: metav1.ObjectMeta{
-	// 		Name:      "agent-cluster-install",
-	// 		Namespace: installConfig.Config.Namespace,
-	// 	},
-	// 	Spec: hiveext.AgentClusterInstallSpec{
-	// 		ClusterDeploymentRef: corev1.LocalObjectReference{
-	// 			Name: installConfig.Config.ObjectMeta.Name,
-	// 		},
-	// 		SSHPublicKey: installConfig.Config.SSHKey,
-	// 		ProvisionRequirements: hiveext.ProvisionRequirements{
-	// 			ControlPlaneAgents: int(*installConfig.Config.ControlPlane.Replicas),
-	// 		},
-	// 	},
-	// }
-
-	// var numberOfWorkers int = 0
-	// for _, compute := range installConfig.Config.Compute {
-	// 	numberOfWorkers = numberOfWorkers + int(*compute.Replicas)
-	// }
-	// agentClusterInstall.Spec.ProvisionRequirements.WorkerAgents = numberOfWorkers
-	// a.Config = agentClusterInstall
-
-	// agentClusterInstallData, err := yaml.Marshal(agentClusterInstall)
-	// if err != nil {
-	// 	return errors.Wrap(err, "failed to marshal agent installer AgentClusterInstall")
-	// }
-
-	// a.File = &asset.File{
-	// 	Filename: agentClusterInstallFilename,
-	// 	Data:     agentClusterInstallData,
-	// }
+	installConfig := &agent.OptionalInstallConfig{}
+	dependencies.Get(installConfig)
 
+	if installConfig.Config != nil {
+		var numberOfWorkers int = 0
+		for _, compute := range installConfig.Config.Compute {
+			numberOfWorkers = numberOfWorkers + int(*compute.Replicas)
+		}
+
+		clusterNetwork := []hiveext.ClusterNetworkEntry{}
+		for _, cn := range installConfig.Config.Networking.ClusterNetwork {
+			_, cidr, err := net.ParseCIDR(cn.CIDR.String())
+			if err != nil {
+				return errors.Wrap(err, "failed to parse ClusterNetwork CIDR")
+			}
+			err = validate.SubnetCIDR(cidr)
+			if err != nil {
+				return errors.Wrap(err, "failed to validate ClusterNetwork CIDR")
+			}
+
+			entry := hiveext.ClusterNetworkEntry{
+				CIDR:       cidr.String(),
+				HostPrefix: cn.HostPrefix,
+			}
+			clusterNetwork = append(clusterNetwork, entry)
+		}
+
+		serviceNetwork := []string{}
+		for _, sn := range installConfig.Config.Networking.ServiceNetwork {
+			cidr, err := ipnet.ParseCIDR(sn.String())
+			if err != nil {
+				return errors.Wrap(err, "failed to parse ServiceNetwork CIDR")
+			}
+			serviceNetwork = append(serviceNetwork, cidr.String())
+		}
+
+		agentClusterInstall := &hiveext.AgentClusterInstall{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      getAgentClusterInstallName(installConfig),
+				Namespace: getObjectMetaNamespace(installConfig),
+			},
+			Spec: hiveext.AgentClusterInstallSpec{
+				ImageSetRef: &hivev1.ClusterImageSetReference{
+					Name: "openshift-4.11",
```

Comment:
I don't think we should hard-code the version number here. I think there are ways to find the version number like: https://github.com/openshift/installer/blob/37ae964e24395c6fc23a349fe39e7037ac66a99d/pkg/asset/releaseimage/default.go#L23

Reply:
Should't be the name of the `ClusterImageSet` resource?

Reply:
(note: in case the current version will be required, I'd recommed to use [version.Version()](https://github.com/openshift/installer/blob/37ae964e24395c6fc23a349fe39e7037ac66a99d/pkg/version/version.go#L52))

Reply:
I agree we shouldn't hard-code it. I think it's fine to set the name to just the name of the cluster. If we're going to put a version then we should go with whatever Andrea suggests :)

Reply:
But either way, we should have a `getClusterImageSetName()` function to get it.
(Currently the ClusterImageSet name is set to `"openshift-" + version.Version()` as Andrea suggested.)

Reply:
ok, so it will get the value as ` openshift-was not built correctly`

Reply:
That happens only when you build from source (and without overriding the release) (ie see also https://github.com/openshift/installer/blob/6f836627e01f868c8ff62e6d19794d76b289d433/pkg/asset/agent/image/releaseimage_test.go#L44)

Reply:
lol, hopefully proper builds will have something more helpful.
Thinking a bit more...
Advantage of using the version: you can see the version just by looking at the AgentClusterInstall, without having to go look at the ClusterImageSet as well.
Advantage of using the cluster name: people will be less tempted to just change the name of the ImageSetRef and expect that to affect the version.

I think I'm tempted to just use the cluster name.